vispy/visuals/volume.py

```diff
@@ -843,11 +843,10 @@ def set_data(self, vol, clim=None, copy=True):
             self._texture.set_clim(clim)
 
         # Apply to texture
-        if should_cast_to_f32(vol.dtype):
-            vol = vol.astype(np.float32)
+        vol = downcast_to_32(vol, copy=copy)
```

Comment:
If using the CPU-scaled texture, don't we not need this downcasting to happen? I mean, if we were given 64-bit floats, the CPU-based scaling would deal with the conversion to a GPU-compatible dtype, right?

Reply:
Uhm, the problem is that the `self._data` is kept for later and used again... If we don't keep a "correct" (downcast) version of the data, then we'll get a million warnings every time the texture needs to be rebuilt (like when clim gets changed). I find it more *correct* to simply hold the data was was actually sent to the gpu, and only warn once.

Reply:
(this is true for both volume and image)

Reply:
> that was actually sent to the gpu

But in the CPU-scaled texture case this casting should never happen and shouldn't be needed so no warning was going to be generated anyway. Now, with this PR, it would produce an unnecessary warning and make a copy of the data.

Reply:
I see. Let's see if tests pass without that casting.

Reply:
Things seem to be doing fine!